blueprint-manager/src/executor/event_handler.rs

```diff
@@ -1,141 +1,40 @@
-       * For each RpcServicesWithBlueprint, fetch the associated gadget binary (fetch/download)
-        -> If the services field is empty, just emit and log inside the executed binary "that states a new service instance got created by one of these blueprints"
-        -> If the services field is not empty, for each service in RpcServicesWithBlueprint.services, spawn the gadget binary, using params to set the job type to listen to (in terms of our old language, each spawned service represents a single "RoleType")
-    */
-
-    let (blueprints, init_event) = if let Some(event) = tangle_runtime.next_event().await {
-        (
-            utils::get_blueprints(&runtime, event.hash, sub_account_id.clone()).await?,
-            event,
-        )
-    } else {
-        return Err(Report::msg("Failed to get initial block hash"));
-    };
-
-    logger.info(format!("Received {} blueprints", blueprints.len()));
-    handle_tangle_event(
-        &init_event,
-        &blueprints,
-        logger,
-        &shell_config,
-        opt,
-        &mut active_shells,
-    )
-    .await?;
-
-    let manager_task = async move {
-        // Step 2: Listen to FinalityNotifications and poll for new/deleted services that correspond to the blueprints above
-        while let Some(event) = tangle_runtime.next_event().await {
-            handle_tangle_event(
-                &event,
-                &blueprints,
-                logger,
-                &shell_config,
-                opt,
-                &mut active_shells,
-            )
-            .await?;
-
-            handle_tangle_block(event, logger, &mut active_shells, &sub_account_id).await;
-        }
-
-        Err::<(), _>(utils::msg_to_error("Finality Notification stream died"))
-    };
-
-    let ctrlc_task = gadget_io::tokio::signal::ctrl_c();
-
-    gadget_io::tokio::select! {
-        res0 = manager_task => {
-            Err(color_eyre::Report::msg(format!("Gadget Manager Closed Unexpectedly: {res0:?}")))
-        },
-
-        _ = ctrlc_task => {
-            logger.info("CTRL-C detected, closing application");
-            Ok(())
-        }
-    }
-}
-
-async fn handle_tangle_block(
-    event: TangleEvent,
+/// Returns true if the blueprint manager needs to update the list of operator-subscribed blueprints
+pub(crate) async fn check_blueprint_events(
+    event: &TangleEvent,
     logger: &DebugLogger,
     active_shells: &mut ActiveShells,
     account_id: &AccountId32,
-) {
+) -> bool {
     let registered_events = event.events.find::<Registered>();
     let unregistered_events = event.events.find::<Unregistered>();
     let service_initiated_events = event.events.find::<ServiceInitiated>();
     let job_called_events = event.events.find::<JobCalled>();
     let job_result_submitted_events = event.events.find::<JobResultSubmitted>();
 
-    // Handle registered events
+    let mut needs_update = false;
+
+    // Handle blueprint registered events
```

Comment:
Doesn't actually seem like these event handlers do anything yet?
- When we register for events we should download the blueprint binary right?
- When we unregister we should note that down (maybe remove binaries but also definitely not process any new services with those blueprints / side-effects to protect against).
- When we see a new service we should spawn the gadget?

Isn't this necessary to test?

Reply:
The logic in the main loop of the work manager recognizes when an update happens, then, it delegates to the appropriate code path. So, yes, we do handle these events albeit, downstream.